src/core/state.rs

```diff
@@ -148,32 +130,27 @@ pub fn insert_asset_definition_v2(
 /// * `storage` A mutable reference to the contract's internal storage.
 /// * `definition` The asset definition to replace by matching on its [asset_type](super::types::asset_definition::AssetDefinitionV2::asset_type)
 /// property.
-pub fn replace_asset_definition_v2(
+pub fn replace_asset_definition_v3(
     storage: &mut dyn Storage,
-    definition: &AssetDefinitionV2,
+    definition: &AssetDefinitionV3,
 ) -> AssetResult<()> {
-    let state = asset_definitions_v2();
-    let key = &definition.storage_key();
-    if let Ok(existing_def) = state.load(storage, key) {
-        // The documentation for the save() function in IndexedMap recommends calling replace() directly after
-        // loading the data, because it's needed for an update and happens internally anyway
-        state
-            .replace(
-                storage,
-                key,
-                definition.to_some(),
-                (&existing_def).to_some(),
-            )
-            .map_err(ContractError::Std)
-    } else {
-        ContractError::RecordNotFound {
-            explanation: format!(
-                "no record exists to update for asset type [{}]",
-                &definition.asset_type
-            ),
-        }
-        .to_err()
-    }
+    let state = ASSET_DEFINITIONS_V3;
+    let key = definition.storage_key();
+    state
+        .update(storage, key, |existing| {
+            if let None = existing {
+                ContractError::RecordNotFound {
+                    explanation: format!(
+                        "no record exists to update for asset type [{}]",
+                        &definition.asset_type
+                    ),
+                }
+                .to_err()
+            } else {
+                definition.to_owned().to_ok()
+            }
+        })
+        .map(|_| ())
```

Comment:
I love rust, but it has some zany-looking syntax lol

Reply:
Haha, yeah, this one got to looking a little weird... not sure why the interface for a Map is so different from an IndexedMap